ZukoAzula2016/src/org/team1540/zukoazula/Shooter.java

```diff
@@ -30,68 +32,74 @@
-
-        BooleanInput shouldSpinUp = setupRollersForSpinup(spinup, actuallyFiring);
-
-        shooter.setup(shouldSpinUp);
-
-        fire.and(shooter.isUpToSpeed).onPress().send(actuallyFiring);
+        
+        StateMachine shooterStates = new StateMachine(0, 
+                "passive", // do nothing
+                "ejecting", // when the ball is being ejected from the intake mechanism
+                "intaking", // when the intake mechanism is intaking a ball
+                "loaded", // when a ball is loaded in the intake mechanism
+                "cocking", // when a ball is being cocked so that it is not in contact w/ the flywheel 
+                "spinup",  // when the flywheel is spinning up to maximum speed
+                "firing"); // when the ball is firing
+        
+        FloatInput flywheelLowSpeed = ZukoAzula.mainTuning.getFloat("Shooter Flywheel Target Low Speed", -80.0f);
+        FloatInput flywheelHighSpeed = ZukoAzula.mainTuning.getFloat("Shooter Flywheel Target High Speed", 2750.0f);
+        FloatInput flywheelTargetVelocity = shooterStates.selectByState(
+                FloatInput.zero, // passive
+                FloatInput.zero, // ejecting
+                flywheelLowSpeed, // intaking
+                FloatInput.zero,
+                FloatInput.zero,
+                flywheelHighSpeed,
+                flywheelHighSpeed);
+        
+        PIDTalon flywheelTalon = new PIDTalon(makeLinkedTalons(), "Shooter Flywheel", flywheelTargetVelocity.withRamping(15.0f, FRC.constantPeriodic));
+        flywheelTalon.setup();
+        
+        BooleanInput inhaleButton = ZukoAzula.controlBinding.addBoolean("Shooter Intake");
+        BooleanInput exhaleButton = ZukoAzula.controlBinding.addBoolean("Shooter Eject");
+        BooleanInput prefireButton = ZukoAzula.controlBinding.addBoolean("Shooter Spinup");
+        BooleanInput fireButton = ZukoAzula.controlBinding.addBoolean("Shooter Fire");
+        BooleanInput cancelShooterButton = ZukoAzula.controlBinding.addBoolean("Shooter Cancel Action");
+        
+        // Behavior
+        shooterStates.setStateWhen("passive", cancelShooterButton.onPress());
+        shooterStates.setStateWhen("ejecting", exhaleButton.onPress());
+        shooterStates.setStateWhen("intaking", inhaleButton.onPress());
+        shooterStates.transitionStateWhen("intaking", "loaded", flywheelTalon.speed.atMost(ZukoAzula.mainTuning.getFloat("Shooter Flywheel Maximum Off Speed", 10.0f)).onPress());
+        shooterStates.setStateWhen("cocking", prefireButton.onPress());
+        shooterStates.setStateWhen("firing", fireButton.onPress()
+                .and(flywheelTalon.speed.atLeast(ZukoAzula.mainTuning.getFloat("Shooter Flywheel Minimum High Speed", 2300.0f))));
+       
+        shooterStates.setStateWhen("passive", 
+                FRC.startDisabled
+                .or(FRC.startTele)
+                .or(FRC.startAuto)
+                .or(FRC.startTest));
```

Comment:
The reason this would have looked better was because it would have been more concise, so spreading it out like this doesn't help comprehension. There are five lines, but no semantic information is added by the extra lines.